### PR TITLE
RELATED: CL-9573 Refactor sorting of metrics

### PR DESCRIPTION
**Pull request by vaclavbohac**
No hashing of items necessary. Lot of `partial` magic removed.

### CODE REVIEW

**Comment by yenkins-admin** [comment]
Test FAILed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/124/

**Comment by yenkins-admin** [comment]
Test FAILed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/133/

**Comment by yenkins-admin** [comment]
Test FAILed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/140/

**Comment by yenkins-admin** [comment]
Test FAILed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/141/

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/142/

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/143/

File: src/execution.js [hunk 1]
```diff
@@ -167,81 +167,93 @@ const getGeneratedMetricIdentifier = (item, aggregation, expressionCreator, hash
...
+    return (type === 'fact' || type === 'attribute' || !allFiltersEmpty(item));
+};
+
+const getDate = date => get(date, 'dateFilter', date);
+
+const getCategories = mdObj => map(mdObj.categories, ({ category }) => category);
+const getFilters = ({ filters }) => filters;
+
+// @TODO This is fucking stupid name
+const getAttribute = mdObj =>
```
Comment (vaclavbohac) [inline src/execution.js]: Maybe it should not be single function. Maybe we should split it in two.

File: src/execution.js [hunk 2]
```diff
@@ -167,81 +167,93 @@ const getGeneratedMetricIdentifier = (item, aggregation, expressionCreator, hash
     return `${type}_${identifier}.generated.${prefix}${aggregation}.${hash}`;
 };
 
-const generatedMetricDefinition = item => {
-    const { title, format, sort } = item;
+const isDerived = (item) => {
```
Comment (vaclavbohac) [inline src/execution.js]: `item` => `measure`

File: src/execution.js [hunk 3]
```diff
@@ -167,81 +167,93 @@ const getGeneratedMetricIdentifier = (item, aggregation, expressionCreator, hash
...
+    const type = get(item, 'type');
+    return (type === 'fact' || type === 'attribute' || !allFiltersEmpty(item));
+};
+
+const getDate = date => get(date, 'dateFilter', date);
+
+const getCategories = mdObj => map(mdObj.categories, ({ category }) => category);
```
Comment (vaclavbohac) [inline src/execution.js]: We can use destructuring in the first param to keep consistency

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/145/

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/146/

**Comment by yenkins-admin** [comment]
Test FAILed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/147/

**Comment by vaclavbohac** [comment]
ok to test

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/148/

File: src/execution.js [hunk 4]
```diff
@@ -167,81 +167,99 @@ const getGeneratedMetricIdentifier = (item, aggregation, expressionCreator, hash
...
+const isDerived = measure => {
+    const type = get(measure, 'type');
+    return (type === 'fact' || type === 'attribute' || !allFiltersEmpty(measure));
+};
+
+const isDateCategory = ({ type }) => type === 'date';
+const isDateFilter = ({ dateFilter }) => dateFilter;
+
+const getCategories = ({ categories }) => map(categories, ({ category }) => category);
+const getFilters = ({ filters }) => filters;
```
Comment (dprentis) [inline src/execution.js]: Just a small thing,  but the discrepancy between isDateCategory and isDateFilter might be confusing - the first accepts the inner object, while the second accepts the outer wrapper. I understand that in the case of filters you need to work with the outer wrappers.

You could however update isDateCategory and getCategories to correspond, eg.
```
const isDateCategory = ({ category }) => category.type === 'date';
const getCategories = ({ categories }) => categories
```
Then you'd need to update usage of getCategories correspondingly - maybe you don't really need getCategories and getFilters in that case.

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/149/

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/150/

**Pull request merged**